src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,307 @@
+
+  private final ApiCallableTransformer apiCallableTransformer = new ApiCallableTransformer();
+  private final ApiMethodTransformer apiMethodTransformer = new ApiMethodTransformer();
+  private final PageStreamingTransformer pageStreamingTransformer = new PageStreamingTransformer();
+  private final PathTemplateTransformer pathTemplateTransformer = new PathTemplateTransformer();
+  private final GapicCodePathMapper pathMapper;
+
+  public GoGapicSurfaceTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME, DOC_TEMPLATE_FILENAME, SAMPLE_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> models = new ArrayList<ViewModel>();
+    GoSurfaceNamer namer = new GoSurfaceNamer(model, apiConfig.getPackageName());
+    List<ServiceDocView> serviceDocs = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(service, apiConfig, createTypeTable(), namer);
+      models.add(generate(context));
+
+      context = SurfaceTransformerContext.create(service, apiConfig, createTypeTable(), namer);
+      models.add(generateExample(context));
+    }
+    models.add(generatePackageInfo(model, apiConfig, namer));
+    return models;
+  }
+
+  private static final ImmutableList<String> PAGE_STREAM_IMPORTS =
+      ImmutableList.<String>of("math;;;");
+
+  private StaticLangXCombinedSurfaceView generate(SurfaceTransformerContext context) {
+    StaticLangXCombinedSurfaceView.Builder view = StaticLangXCombinedSurfaceView.newBuilder();
+
+    SurfaceNamer namer = context.getNamer();
+    Model model = context.getModel();
+    Interface service = context.getInterface();
+    ApiConfig apiConfig = context.getApiConfig();
+
+    view.templateFileName(XAPI_TEMPLATE_FILENAME);
+    view.serviceDoc(doc(service));
+    view.localPackageName(namer.getLocalPackageName());
+    view.clientTypeName(namer.getApiWrapperClassName(service));
+    view.clientConstructorName(namer.getApiWrapperClassConstructorName(service));
+    view.defaultClientOptionFunctionName(namer.getDefaultApiSettingsFunctionName(service));
+    view.defaultCallOptionFunctionName(namer.getDefaultCallSettingsFunctionName(service));
+    view.callOptionsTypeName(namer.getCallSettingsTypeName(service));
+    view.serviceOriginalName(model.getServiceConfig().getTitle());
+    view.servicePhraseName(namer.getServicePhraseName(service));
+    view.grpcClientTypeName(namer.getGrpcClientTypeName(service));
+    view.grpcClientConstructorName(namer.getGrpcClientConstructorName(service));
+
+    String outputPath = pathMapper.getOutputPath(service, apiConfig);
+    String fileName = GoSurfaceNamer.getReducedServiceName(service) + "_client.go";
```

Comment:
I missed this before - access getReducedServiceName through SurfaceNamer instead of directly from GoSurfaceNamer.